### PR TITLE
add admins as member on check

### LINKED ISSUE #2530: Excempt adminLogged and dev steam64s from the member radius
This makes no sense to happen when I'm logged in as admin and teleporting around to check dev stuff.
image

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
    admins were not considered members for some reason so i
    made the isMember check add admins as members if they are not already one

### Please specify which Issue this PR Resolves.
closes #2530 

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/functions/OrgPlayers/fn_isMember.sqf [hunk 1]
```diff
@@ -3,5 +3,17 @@ if !(membershipEnabled) exitWith {true};
...
 // Remote-controlled AI unit is not a member
 if (_player getVariable ["owner", _player] != _player) exitWith {false};
+
 // Original player unit can be a member
-_player getVariable ["A3A_playerUID", getPlayerUID _player] in membersX;
+private _uid = _player getVariable ["A3A_playerUID", getPlayerUID _player]
+private _isMember = _uid in membersX;
+
+//if admin and not member, add as temp member
+if !(_isMember && {call A3A_fnc_getAdmin isEqualTo _player}) then {
```
Comment (killerswin2) [inline A3A/addons/core/functions/OrgPlayers/fn_isMember.sqf:12]: your code has an edge case, you forgot about De Morgan's laws. Move the negation to the isMember variable.

**Review by killerswin2 — CHANGES_REQUESTED** [review]

**Review by killerswin2 — APPROVED** [review]
that works. look good.

**Comment by jaj22** [comment]
Bit wasteful (it's making isMember much slower on each call by a guest), and it has publicVariable fighting risks with the (bad) member escalation code in initClient. Could use the onUserAdminStateChanged mission event handler instead which solves both problems, although the initClient stuff could also do with a server-side replacement.

File: A3A/addons/core/functions/init/fn_initServer.sqf [hunk 2]
```diff
@@ -176,6 +176,17 @@ if (isPlayer A3A_setupPlayer) then {
     membersX pushBackUnique getPlayerUID A3A_setupPlayer;
     theBoss = A3A_setupPlayer; publicVariable "theBoss";
 };
+
+//add admin as member if not on loggin
+addMissionEventHandler ["OnUserAdminStateChanged", {
+    params ["_networkId", "_loggedIn", "_votedIn"];
+    private _uid = (getUserInfo _networkId)#2;
+    if !(_uid in membersX) then {
+        membersX pushBackUnique (getUserInfo _networkId)#2;
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initServer.sqf:185]: Should probably have been `membersX pushBack _uid` but it'll work.

**Review by jaj22 — APPROVED** [review]

**Pull request merged**